src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -76,7 +77,44 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
 
                         string challengeData = challenge.ChallengeData;
 
-                        string spn = "HTTP/" + authUri.IdnHost;
+                        // Calculate SPN (Service Principal Name) using the host name of the request.
+                        // Use the request's 'Host' header if available. Otherwise, use the request uri.
+                        // Ignore the 'Host' header if this is proxy authentication since we need to use
+                        // the host name of the proxy itself for SPN calculation.
+                        string hostName;
+                        if (!isProxyAuth && request.HasHeaders && request.Headers.Host != null)
```

Comment:
@davidsh, IIRC, there had been a couple of changes to this code across a few PRs.  It looks like this includes all such changes as is appropriate, but just want to confirm.

Reply:
Yes, I went thru all the various PRs and subsequent PRs to make sure that the fixes are scoped and appropriate.